test/intersections.jl

```diff
@@ -150,6 +150,38 @@
     @inferred someornone(s1, s2)
   end
 
+  @testset "Rays" begin
+    # rays in 2D
+    r₁ = Ray(P2(1,0), V2(2,1))
+    r₂ = Ray(P2(0,2), V2(2,-3))
+    r₃ = Ray(P2(0.5,1), V2(1,-2))
+    r₄ = Ray(P2(0,2), V2(1,-3))
+    r₅ = Ray(P2(4,1.5), V2(4,2))
+    r₆ = Ray(P2(2,0.5), V2(-0.5, -0.25))
+    r₇ = Ray(P2(4,0), V2(0,1))
+    @test intersection(r₁, r₂) |> type == CrossingRays #CASE 1
+    @test r₁ ∩ r₂ == P2(1.25,0.125)
+    @test r₁ ∩ r₇ == P2(4, 1.5)
+    @test intersection(r₁, r₃) |> type == MidTouchingRays #CASE 2
+    @test r₁ ∩ r₃ == r₁.p #origin of first ray
+    @test r₅ ∩ r₇ == r₅.p
```

Comment:
We never access fields of structs directly. We can only use the exported API, in this case origin(ray) gives the origin point.